lib/dealership.rb

```diff
@@ -0,0 +1,65 @@
+class Dealership
+  attr_reader :inventory
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def inventory_count
+    @inventory.count
+  end
+
+  def has_inventory?
+    inventory_count >= 1
+  end
+
+  def cars_by_make(make)
+    cars_by_make = []
+    @inventory.each do |car|
+      if car.make == make
+        cars_by_make << car
+      end
+    end
+    cars_by_make
+  end
+
+  def total_value
+    total_value = 0
+    @inventory.each do |car|
+      total_value += car.total_cost
+    end
+    total_value
+  end
+
+  def details
+    details = {}
+    details["total_value"] = total_value
+    details["address"] = @address
+    details
+  end
+
+  def average_price_of_car
+    average_price = total_value / inventory_count
+    average_price.to_s.insert(-4, ",")
+  end
+
+  def cars_sorted_by_price
+    inventory.sort_by do |car|
+      car.total_cost
+    end
+  end
+
+  def inventory_hash
+    inventory_hash = {}
+    inventory.each do |car|
+      inventory_hash[car.make] = car
+    end
+    inventory_hash
+  end
```

Comment:
This is very close. Can you figure out why this does not work as described in the interaction pattern?

Reply:
When `.each` is iterating through `inventory`, it is assigning the current car as the value for the `car.make` key inside of the hash. However, if that key already has a value assigned it is re-assigning the key to the new value instead of adding it to the key in an array. This is making the return value of the method only return the last car to be assigned to each unique key instead of all of the cars that were added to the keys.  If I had more time, I think I could have figured this one out, but that was the hang-up I kept running into.

Reply:
I took some time on Friday and ended up figuring this problem out. I've pushed new changes of it to Github if you're interested in checking it out.